content/en/blog/_posts/2021-10-05-nsa-cisa-hardening.md

```diff
@@ -0,0 +1,413 @@
+2. [gvisor](https://gvisor.dev/docs/user_guide/containerd/quick_start/)
+
+As discussed here and in the guidance, many features and tooling in and around
+Kubernetes tooling exists that can enhance the isolation boundaries between two
+pods. Based on relevant threats and risk posture, users should pick and choose
+between them, instead of trying to apply all. Having said that, cluster
+level isolation remains the strictest workload isolation mechanism, inspite of
+improvements mentioned earlier here and in the guide.
+
+## Network Separation and Hardening
+
+Kubernetes Networking can be tricky and this section focuses on how to secure
+and harden the relevant configurations. The guide identifies the following as key
+takeaways: 
+- Using network policies to create isolation between resources, 
+- Securing control plane
+- Encrypting traffic and sensitive data
+
+### Network Policies
+
+Network policies can be created with the help of network plugins. In order to
+make the creation and visualization easier for users, Cilium supports
+a [web GUI tool](https://editor.cilium.io). It creates Kubernetes Network
+policies as well as Cilium Network policies, which can be used to minimize the
+attack vector.
+
+Another scenario that is worth exploring is the usage of external IPs. Some
+services when misconfigured can create random external IPs. An attacker can take
+advantage of this configuration and easily intercept traffic. This vulnerability
+has been reported
+in [CVE-2020-8554](https://www.cvedetails.com/cve/CVE-2020-8554/).
+Using [externalip-webhook]([https://github.com/kubernetes-sigs/externalip-webhook)
+can mitigate this vulnerability. It will prevent the services from using random
+external IPs. [Webhook](https://github.com/kubernetes-sigs/externalip-webhook) 
+will only allow creation of services which doesn't require external IP or whose 
+external IPs are within the range specified by the administrator.
+
+> CVE-2020-8554 - Kubernetes API server in all versions allow an attacker who is able to create a ClusterIP service and set the spec.externalIPs field, to intercept traffic to that IP address. Additionally, an attacker who is able to patch the status (which is considered a privileged operation and should not typically be granted to users) of a LoadBalancer service can set the status.loadBalancer.ingress.ip to similar effect.
+
+### Resource Policies
+
+In addition to adding Resource Quotas and limits, consider adding PID limits and
+restrictions, to avoid PID exhaustion. More details can be
+found [here](https://kubernetes.io/docs/concepts/policy/pid-limiting/).
+
+### Control Plane Hardening
+
+In the next section, the guide covers control plane hardening. It is worth
+noting that
+from [Kubernetes 1.20](https://github.com/kubernetes/kubernetes/issues/91506),
+insecure port has been removed.
+
+### Etcd
+
+As a general thumb of rule, the etcd server should be configured to only trust
```

Comment:
Nit pick: `As a general rule of thumb`

Reply:
Better to not include "of thumb" at all, as not all readers are hip to that term :)